mitodl/open-discussions
# adjust post overflow menu contents

#### Pre-Flight checklist


#### What are the relevant tickets?

closes #1939
closes #1938 

#### What's this PR do?

this basically

- hides the 'remove' button from the overflow menu on the user's own posts on the post detail page
- does the same on the channel page
- adds the 'delete' option to the user's own posts on the channel page
- adds the 'delete' option to the user's own posts on the home page

#### How should this be manually tested?

make sure that the above description is accurate! as a moderator user you should no longer see the 'remove' / 'approve' options for your own posts. you should now see the 'delete' option for your own posts wherever they may appear.

other stuff should keep working.

#### Screenshots (if appropriate)

![deletebutton](https://user-images.githubusercontent.com/6207644/55565628-2e323100-56c8-11e9-9295-fef8eec941d5.png)
